packages/text-field/src/TextField/TextField.tsx

```diff
@@ -0,0 +1,135 @@
+import { useComposedRefs } from "@telegraph/compose-refs";
+import { Icon } from "@telegraph/icon";
+import { clsx } from "clsx";
+import React from "react";
+
+import { COLOR, SIZE } from "./TextField.constants";
+
+type TextFieldProps = Omit<
+  React.HTMLAttributes<HTMLInputElement>,
+  "value" | "onChange"
+> & {
+  size?: "1" | "2" | "3";
+  variant?: "outline" | "ghost";
+  disabled?: boolean;
+  error?: boolean | string;
+  value?: string;
+  onChange?: (
+    value: string,
+    { event }: { event: React.ChangeEvent<HTMLInputElement> },
+  ) => void;
```

Comment:
Hm, this is different than typical input/form handlers with an event as the first arg. Any reason for the divergence here?

Reply:
99% of the time all you want is `event.target.value` so this makes it easier to grab that. It also helps for when we add a custom Combobox to `Field` where there won't be a ChangeEvent, so then there is just a similar value that can be used across both.

Reply:
I'm just concerned with handling non-native inputs like a combobox where there is no event being returned, this provides some consistency. But you're probably right that this shouldn't be happening in `TextField` but maybe rather `Field`.